differential-expression/rnaseq_DGE.Rmd

```diff
@@ -0,0 +1,216 @@
+  dplyr::mutate(mutation_status = stringr::word(refinebio_title,
+                                                -1, sep = "-")) %>%
+  # now let's make this simpler
+  dplyr::mutate(mutation_status = dplyr::case_when(
+    grepl("ASXL1|ASXL2", mutation_status) ~ "mutation",
+    grepl("ASXLwt", mutation_status) ~ "wildtype"
+  )) 
+```
+
+Let's take a look at `metadata.df` to see if this worked.
+
+```{r}
+# looking at the first 6 rows of the metadata_df and only at the columns that
+# contain the title and the mutation status we extracted from the title
+head(dplyr::select(metadata.df, refinebio_title, mutation_status))
+```
+
+## Differential expression analysis
+
+Now that we've prepped our data, we're ready to test for differentially 
+expressed genes.
+
+Let's make sure samples are in the same order.
+
+```{r}
+all.equal(colnames(expression.df), metadata.df$refinebio_accession_code)
+```
+
+```{r}
+# put samples in ascending orderaccording to their accession code in the 
+# metadata
+metadata.df <- metadata.df %>%
+  dplyr::arrange(refinebio_accession_code)
+
+# use the metadata.df information to reorder the columns of the expression
+# data frame
+expression.df <- expression.df[, metadata.df$refinebio_accession_code]
+```
+
+Check again.
+
+```{r}
+all.equal(colnames(expression.df), metadata.df$refinebio_accession_code)
+```
+
+### limma-voom
+
+Here, we're following the steps in [this section](https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html#limma-voom) 
+of the [tximport vignette](https://bioconductor.org/packages/release/bioc/vignettes/tximport/inst/doc/tximport.html).
+
+```{r}
+# construct a DGEList object from the refine.bio output
+y <- DGEList(as.matrix(expression.df))
+
+# filtering
+keep <- filterByExpr(y)
+y <- y[keep, ]
+
+# edgeR calculation of normalization factors to scale raw library sizes
+y <- calcNormFactors(y)
+
+# design matrix that captures the mutated vs. wild type comparison
+design <- model.matrix(~ metadata.df$mutation_status)
+
+# this is the step that transforms data to log2CPM, estimates the mean-variance
+# relationship and computes observation-level weights
+v <- voom(y, design)  # ready for lmFit
+```
+
+```{r Apply linear model}
+# Apply linear model to data
+fit <- lmFit(v, design = design)
+
+# Apply empirical Bayes to smooth standard errors
+fit <- eBayes(fit)
+
+# Apply multiple testing correction and obtain stats
+stats <- topTable(fit, number = nrow(expression.df)) %>% 
+  tibble::rownames_to_column("Gene")
+```
+
+Write the `topTable` output to file.
+
+```{r  Write results}
+readr::write_tsv(stats, file.path("results", "SRP078441_limma_results.tsv"))
+```
+
+### Volcano plot
+
+```{r}
+# Create volcano plot 
+voc.plot <- volcanoplot(fit, main = "Volcano Plot SRP078441")
+
+# Save plot to png
+png(file.path("plots", "volcano_plot_SRP078441.png"))
+volcanoplot(fit, main = "Volcano Plot SRP078441")
+dev.off()
+
+# Print out the plot here
```

Comment:
This printing out part that doesn't actually print out is my bad but I think it will also be unnecessary here.